infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java

```diff
@@ -31,8 +35,53 @@
     public static final String PREFIX_EPOCH = "SERVER_EPOCH";
     public static final String KEY_EPOCH = "CURRENT";
 
-    ExecutorService handlerWorkers = Executors.newWorkStealingPool(Runtime.getRuntime()
-            .availableProcessors());
+    public static class ServerThreadFactory
+            implements ForkJoinPool.ForkJoinWorkerThreadFactory {
+
+        public static final String THREAD_PREFIX = "ServerRouter-";
+        final AtomicInteger threadNumber = new AtomicInteger(0);
+
+        public static class ServerWorkerThread extends ForkJoinWorkerThread {
+
+            protected ServerWorkerThread(final ForkJoinPool pool, final String threadName) {
+                super(pool);
+                this.setName(threadName);
+                this.setUncaughtExceptionHandler(NettyServerRouter::handleUncaughtException);
+            }
+
+            @Override
+            protected void onTermination(Throwable exception) {
+                if (exception != null) {
+                    log.error("onTermination: Thread terminated due to {}:{}",
+                            exception.getClass().getSimpleName(),
+                            exception.getMessage(),
+                            exception);
+                } else {
+                    log.warn("onTermination: Thread terminated (completed normally).");
+                }
+                super.onTermination(exception);
+            }
+        }
+
+        @Override
+        public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+            return new ServerWorkerThread(pool,
+                    THREAD_PREFIX + threadNumber.getAndIncrement());
+        }
+    }
+
+    protected static void handleUncaughtException(Thread t, @Nonnull Throwable e) {
+        log.error("handleUncaughtException[{}]: Uncaught {}:{}",
+                t.getName(),
+                e.getClass().getSimpleName(),
+                e.getMessage(),
+                e);
+    }
+
+    final protected ExecutorService handlerWorkers =
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)